lib/providers/users.dart

```diff
@@ -27,27 +27,40 @@ class Users with ChangeNotifier {
   void updateUser(int id, String name) {
     _users.update(
       id,
-      (oldUser) => User(id: oldUser.id, name: name),
+      (oldUser) => User(id: oldUser.id, name: name, isCurrentUser: oldUser.isCurrentUser),
     );
     update(table: 'users', object: _users[id]);
     notifyListeners();
   }
 
-  Future<int> addUser(String name) async {
-    User user = User(name: name);
+  Future<int> addUser(String name, {bool isCurrentUser=false}) async {
```

Comment:
so we are allowing mulitple users?  thought this was a 'nice to have' feature?

Reply:
Users in this context includes contacts, when looking at the structure users and contacts had the same columns and uses so I made them the same thing. Only one current user is allowed and is used to identify which side of the screen the messages show up on. It also allows multiple users to be implemented easily but that's not really the point of it.

Reply:
oh okay! so how are we linking the current user to thier contacts (link back to the user table)? do we have a many to many linking?

Reply:
Any other user in the device is considered a contact and messages are linked to users so that we know who it is from and which conversation(s) they belong to

Reply:
Lol... didn’t even think of that.... i keep thinking in terms of a server db not app.... makes sense. Thanks! :)

Reply:
Its been super hard separating the two, when you only have one user things don't need to be as strict.